### PR TITLE
Add integration tests for client namespace

### PR DESCRIPTION
**Pull request by cddr**
Adds a bunch of tests for the client namespace. This PR covers the following aspects.

  - basic consumer/producer creation and teardown
  - send!, produce!
  - topic subscription and assignment
  - topic partition interrogation

There's still a bit more that remains without tests like offset setting/getting but these features are used by the test-machine's own test-suite.

### CODE REVIEW

File: src/jackdaw/client.clj [hunk 1]
```diff
@@ -305,3 +301,8 @@
                         (map (fn [^PartitionInfo x]
                                (TopicPartition. (.topic x) (.partition x)))))]
     (apply assign consumer partitions)))
+
+(defn assignment
+  "Return the assigned topics and partitions of a consumer."
+  [^KafkaConsumer consumer]
+  (map jd/datafy (.assignment consumer)))
```
Comment (creese) [inline src/jackdaw/client.clj]: Why did you move this to the bottom?

File: test/jackdaw/client_test.clj [hunk 2]
```diff
@@ -1,12 +1,85 @@
...
+            [jackdaw.test.serde :as serde]
+            [jackdaw.data :as data])
   (:import [java.util.concurrent LinkedBlockingQueue TimeUnit]
            java.time.Duration
            [org.apache.kafka.clients.consumer Consumer ConsumerRecord ConsumerRecords]
            [org.apache.kafka.clients.producer Producer]
            org.apache.kafka.common.TopicPartition))
 
+(def foo-topic
+  (serde/resolver {:topic-name "foo"
```
Comment (creese) [inline test/jackdaw/client_test.clj]: Not a blocker, but we should refactor this to use `jackdaw.serdes.resolver/serde-resolver` after https://github.com/FundingCircle/jackdaw/pull/71 is merged.

File: src/jackdaw/client.clj (hunk 1 above)
Comment (kidpollo) [reply src/jackdaw/client.clj]: dont exactly recall doing that :P was not intentional

**Comment by kidpollo** [comment]
YAY THANKS!

File: src/jackdaw/client.clj (hunk 1 above)
Comment (cddr) [reply src/jackdaw/client.clj]: To keep it together with the function that lets you set the assignments.
